PS4k/src/com/vitco/layout/content/menu/MainMenuLogic.java

```diff
@@ -730,6 +785,206 @@ protected Object doInBackground() throws Exception {
+                                            bgColor = Color.decode(colorHex.startsWith("#") ? colorHex : "#" + colorHex);
+                                        else throw new NumberFormatException();
+                                    }
+                                } catch (NumberFormatException e) {
+                                    console.addLine("Error: Invalid image background color");
+                                    errorHandler.handle(e);
+                                    return false;
+                                }
+
+                                final long time = System.currentTimeMillis();
+
+                                // exported files count
+                                int count = 0;
+                                for (float i = 0f; i < 360f; i += animFramesCount) {
+                                    // get current rotation angle in radians (clockwise or anti-clockwise)
+                                    final float rotAngleDeg = (float) Math.toRadians((animClockwise ? i : 360f - i));
+                                    // current export file name
+                                    final String renderName = baseName + "_" + (count + 1) + ".png";
+                                    progressDialog.setActivity("Exporting " + renderName, true);
+
+                                    // camera's rotation calculation
+                                    SimpleVector camPos = new SimpleVector(initCamPos);
+                                    camPos.x += (float) Math.sin(rotAngleDeg);
+                                    camPos.z += (float) Math.cos(rotAngleDeg);
+
+                                    try {
+                                        BufferedImage image = mainView.getImage(camPos.rotate(new SimpleVector(0f,
+                                                rotAngleDeg, 0f)), bgColor, addWatermark, true);
+                                        ImageIO.write(image, "png", new File(renderName));
+                                        count++;
+                                    } catch (Exception e) {
+                                        errorHandler.handle(e);
+                                        console.addLine(String.format(langSelector.getString("export_files_error"), count));
+                                        return false;
+                                    }
+                                }
+                                console.addLine(String.format(langSelector.getString("export_files_successful"), System.currentTimeMillis() - time, count));
+                                return null;
+                            }
+                        });
+
+                        // ===========
+
+                    } else if (dialog.is("export_type=turntable_animation")) {
+
+                        // ===========
+
+                        // -- export turntable animation
```

Comment:
Yes. I'm thinking all the exporters will be separated out eventually. But for now just doing it with this one is fine. I really need to rework the `MainMenuLogic` class. It's a giant mess :)